### PR TITLE
Download session & subject archives with new API

### PR DESCRIPTION
**Pull request by anastasiiadidan**
Should be merged with https://github.com/stanfordnmbl/opencap-api/pull/70

### CODE REVIEW

File: src/components/pages/Step5.vue [hunk 1]
```diff
@@ -229,12 +229,70 @@
...
+                            <v-col cols="2">
+                                <v-icon x-large color="green">mdi-download</v-icon>
+                            </v-col>
+                            <v-col cols="10">
+                                <p v-if="isArchiveInProgress & !isArchiveDone">
+                                    <v-progress-circular  indeterminate class="mr-2" color="grey" size="14" width="2" />
+                                    Download in progress...
+                                </p>
+                                <p v-if="!(isArchiveInProgress || isArchiveDone)">
+                                    Do you want to download all data associated to the
```
Comment (antoinefalisse) [inline src/components/pages/Step5.vue:253]: Rephrase: Do you want to download all the data from session <>

File: src/components/pages/Step5.vue [hunk 2]
```diff
@@ -229,12 +229,70 @@
...
+                                    session <code>{{session.id}}</code>?
+                                    (This includes every trial associated to it, and can take several minutes).
```
Comment (antoinefalisse) [inline src/components/pages/Step5.vue:255]: Rephrase: (This can take several minutes)

File: src/components/pages/Step5.vue [hunk 3]
```diff
@@ -229,12 +229,70 @@
...
+                                </p>
+                                <p v-if="isArchiveDone">
+                                    The archive has been generated successfully!
```
Comment (antoinefalisse) [inline src/components/pages/Step5.vue:258]: Rephrase: The archive has been successfully generated. Click on data.zip to download.

File: src/components/pages/Step5.vue [hunk 4]
```diff
@@ -229,12 +229,70 @@
...
+                                @click="showArchiveDialog = false;"
+                            >
+                                Cancel
+                            </v-btn>
+                            <v-btn 
+                                v-if="isArchiveDone"
+                                :href="archiveUrl"
+                                :download="archiveName"
+                            >
+                                {{archiveName}}
```
Comment (antoinefalisse) [inline src/components/pages/Step5.vue:277]: Could we just use data.zip for the button, the full archive name is complicated and there is no confusion possible so no need to incorporate the session ID in the button name.

File: src/components/pages/Subjects.vue [hunk 5]
```diff
@@ -223,7 +223,67 @@
...
+                                    <v-progress-circular  indeterminate class="mr-2" color="grey" size="14" width="2" />
+                                    Download in progress...
+                                  </p>
+                                  <p v-if="!(isArchiveInProgress || isArchiveDone)">
+                                    Do you want to download all data associated to the
+                                    session <code>{{item.name}}</code>?
+                                    (This includes every trial associated to it, and can take several minutes).
+                                  </p>
+                                  <p v-if="isArchiveDone">
+                                    The archive has been generated successfully!
```
Comment (antoinefalisse) [inline src/components/pages/Subjects.vue:252]: Rephrase: The archive has been successfully generated. Click on data.zip to download.

File: src/components/pages/Subjects.vue [hunk 6]
```diff
@@ -223,7 +223,67 @@
...
+                                >
+                                    Cancel
+                                </v-btn>
+                                
+                                <v-btn 
+                                    v-if="isArchiveDone"
+                                    :href="archiveUrl"
+                                    :download="archiveName"
+                                >
+                                    {{archiveName}}
```
Comment (antoinefalisse) [inline src/components/pages/Subjects.vue:272]: Could we just use data.zip for the button, the full archive name is complicated (eg, users do not know about the index of a subject) and there is no confusion possible so no need to incorporate the user ID in the button name.

**Review by antoinefalisse — CHANGES_REQUESTED** [review]
@anastasiiadidan, this is looking great. I left some minor comments. Thanks.

**Review by olehkorkh-planeks — APPROVED** [review]

**Review by antoinefalisse — APPROVED** [review]

**Review by antoinefalisse — APPROVED** [review]

**Pull request merged**